[Merge-Queue] Add author to commit message

JonWBedard opened the pull request:
#### 1e8eb307daa055dde54e7c11f60d4153ac9bd325
<pre>
[Merge-Queue] Add author to commit message
<a href="https://bugs.webkit.org/show_bug.cgi?id=238752">https://bugs.webkit.org/show_bug.cgi?id=238752</a>
&lt;rdar://problem/91254339 &gt;

Reviewed by Aakash Jain.

svn.webkit.org will attribute a change to a specific author
if &quot;Patch by&quot; is in the commit message.

* Tools/CISupport/ews-build/factories.py:
(MergeQueueFactory.__init__): Add AddAuthorToCommitMessage step.
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps): Ditto.
* Tools/CISupport/ews-build/master.cfg: Determine commit author from github.head.user.login.
* Tools/CISupport/ews-build/steps.py:
(AddAuthorToCommitMessage):
(AddAuthorToCommitMessage.__init__):
(AddAuthorToCommitMessage.author): Return commit author&apos;s name and email, falling back
to the committers name and email if those cannot be determined.
(AddAuthorToCommitMessage.start): Insert &quot;Patch by&quot; into commit message.
(AddAuthorToCommitMessage.getResultSummary):
(AddAuthorToCommitMessage.doStepIf): Only do step if the author can be determined.
(AddAuthorToCommitMessage.hideStepIf): Hide step if skipped.
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/249265@main">https://commits.webkit.org/249265@main</a>
</pre>

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -4727,6 +4727,69 @@ def hideStepIf(self, results, step):
+
+    def getResultSummary(self):
+        if self.results == FAILURE:
+            return {'step': 'Failed to add author'}
```
This message can be more descriptive. failed to add author, where?

JonWBedard committed:
[Merge-Queue] Add author to commit message
https://bugs.webkit.org/show_bug.cgi?id=238752
<rdar://problem/91254339>

Reviewed by Aakash Jain.

svn.webkit.org will attribute a change to a specific author
if "Patch by" is in the commit message.

* Tools/CISupport/ews-build/factories.py:
(MergeQueueFactory.__init__): Add AddAuthorToCommitMessage step.
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps): Ditto.
* Tools/CISupport/ews-build/master.cfg: Determine commit author from github.head.user.login.
* Tools/CISupport/ews-build/steps.py:
(AddAuthorToCommitMessage):
(AddAuthorToCommitMessage.__init__):
(AddAuthorToCommitMessage.author): Return commit author's name and email, falling back
to the committers name and email if those cannot be determined.
(AddAuthorToCommitMessage.start): Insert "Patch by" into commit message.
(AddAuthorToCommitMessage.getResultSummary):
(AddAuthorToCommitMessage.doStepIf): Only do step if the author can be determined.
(AddAuthorToCommitMessage.hideStepIf): Hide step if skipped.
* Tools/CISupport/ews-build/steps_unittest.py:
